### PR TITLE
Port master to humble

### PR DESCRIPTION
**Pull request by livanov93**
# Description

Move `master` branch features to `humble` branch.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,19 +1,27 @@
 # ign_ros2_control
 
+ROS2 Distro | Build Status | Package build |
+:---------: | :----: | :----------: |
+[![Licence](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0) |  [![Build Status](http://build.ros2.org/buildStatus/icon?job=Gdev__ign_ros2_control__ubuntu_focal_amd64)](http://build.ros2.org/job/Gdev__ign_ros2_control__ubuntu_focal_amd64) |  [![Build Status](http://build.ros2.org/buildStatus/icon?job=Gbin_uF64__ign_ros2_control__ubuntu_focal_amd64__binary)](http://build.ros2.org/job/Gbin_uF64__ign_ros2_control__ubuntu_focal_amd64__binary) |
+
```
Comment (ahcorde) [inline README.md:6]:
```suggestion
ROS2 Distro | Build Status | Package build |
:---------: | :----: | :----------: |
[![Licence](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0) |  [![Build Status](http://build.ros2.org/buildStatus/icon?job=Hdev__gz_ros2_control__ubuntu_focal_amd64)](http://build.ros2.org/job/Hdev__gz_ros2_control__ubuntu_focal_amd64) |  [![Build Status](http://build.ros2.org/buildStatus/icon?job=Hbin_uF64__gz_ros2_control__ubuntu_focal_amd64__binary)](http://build.ros2.org/job/Hbin_uF64__gz_ros2_control__ubuntu_focal_amd64__binary) |

```

File: ign_ros2_control/src/ign_system.cpp [hunk 2]
```diff
@@ -505,6 +631,96 @@ hardware_interface::return_type IgnitionSystem::write(
...
+
+      if (vel == nullptr) {
+        this->dataPtr->ecm->CreateComponent(
+          this->dataPtr->joints_[i].sim_joint,
+          ignition::gazebo::components::JointVelocityCmd({target_vel}));
+      } else if (!vel->Data().empty()) {
+        vel->Data()[0] = target_vel;
+      } else if (!vel->Data().empty()) {
+        vel->Data()[0] = target_vel;
+      }
```
Comment (ahcorde) [inline ign_ros2_control/src/ign_system.cpp:649]: I think we can remove this block

```suggestion
      } else if (!vel->Data().empty()) {
        vel->Data()[0] = target_vel;
      }
```

**Review by ahcorde — CHANGES_REQUESTED** [review]
I did a first pass, I will test it

File: Dockerfile/Dockerfile [hunk 3]
```diff
@@ -1,7 +1,8 @@
-FROM ubuntu:20.04
+FROM ubuntu:22.04
 
 ENV DEBIAN_FRONTEND noninteractive
 ENV IGNITION_VERSION fortress
+ENV ROS_DISTRO rolling
```
[the remark this answers is not in the document]
Comment (bmagyar) [reply Dockerfile/Dockerfile:5]:
```suggestion
ENV ROS_DISTRO humble
```

File: ign_ros2_control/src/ign_system.cpp (hunk 2 above)
Comment (livanov93) [reply ign_ros2_control/src/ign_system.cpp:647]: [2568e8c](https://github.com/ros-controls/gz_ros2_control/pull/121/commits/2568e8cf443f0e29eecb707df23703599e1621fa)

**Comment by livanov93** [comment]
@ahcorde Please can you take another look and test examples?

**Review by ahcorde — APPROVED** [review]

**Pull request merged**